packages/legend-studio/src/components/editor/side-bar/LocalChanges.tsx

```diff
@@ -18,15 +18,101 @@ import { observer } from 'mobx-react-lite';
 import { clsx, PanelLoadingIndicator } from '@finos/legend-art';
 import { EntityDiffViewState } from '../../../stores/editor-state/entity-diff-editor-state/EntityDiffViewState';
 import { EntityDiffSideBarItem } from '../../editor/edit-panel/diff-editor/EntityDiffView';
-import { FaInfoCircle, FaDownload } from 'react-icons/fa';
+import { FaInfoCircle, FaDownload, FaUpload, FaTimes } from 'react-icons/fa';
 import { MdRefresh } from 'react-icons/md';
 import { GoSync } from 'react-icons/go';
 import { LEGEND_STUDIO_TEST_ID } from '../../LegendStudioTestID';
 import { flowResult } from 'mobx';
-import type { EntityDiff } from '@finos/legend-server-sdlc';
+import type { EntityChange, EntityDiff } from '@finos/legend-server-sdlc';
 import { entityDiffSorter } from '../../../stores/EditorSDLCState';
 import { useEditorStore } from '../EditorStoreProvider';
 import { useApplicationStore } from '@finos/legend-application';
+import { Dialog } from '@material-ui/core';
+
+const PatchModalEditor = observer(() => {
```

Comment:
Can we use something like [react-dropzone](https://www.npmjs.com/package/react-dropzone)?

Reply:
Also let's rename this to `PatchLoader`